lib/devos/mkFlake.nix

```diff
@@ -0,0 +1,198 @@
+{ lib, flake-utils, deploy }:
+let
+  mkFlake = { self, ... } @ args:
+      # TODO: pipe args through importers
+    mkFlake' args;
+
+  mkFlake' =
+    {
+      self
+      , name
+      , nixos
+      , overlay
+      , preOverlays ? [ ]
+      , systems ? [ "x86_64-linux" ]
+      , shell ? null
+      , config ? { allowUnfree = true; }
+
+      # internal & external module declarations
+      , nixosModules ? [ ]
+      , extraNixosModules ? [ ]
+      , extraSpecialArgs ? { }
+      , homeModules ? [ ]
+      , extraHomeModules ? [ ]
+      , extraHomeSpecialArgs ? { }
+      # , devShellModules
+
+      # applied devos environment configuriguration
+      , profiles
+      , users
+      , userProfiles
+      , suites
+      , hosts
+
+      # backports from alternate channel
+      , nixos' ? null
```

Comment:
I called it `nixos'`: it's `nixos` but different. `'` is not necesarily an indicator of "different", but at least it's an indicator of something that was transformed by something and now is a derivation of it. So we could say, the original `nixos` was transformed by a series of updates to become `nixos'`.

I felt override did not communicate enough what's in the box.